Exiled.API/Features/Core/Generic/StaticActor.cs

```diff
@@ -0,0 +1,169 @@
+// -----------------------------------------------------------------------
+// <copyright file="StaticActor.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Features.Core.Generic
+{
+    using Exiled.API.Features;
+    using Exiled.API.Features.Core;
+
+    /// <summary>
+    /// This is a generic Singleton implementation for components.
+    /// <br>Create a derived class where the type <typeparamref name="T"/> is the script you want to "Singletonize"</br>
+    /// </summary>
+    /// <typeparam name="T">The type of the class.</typeparam>
+    /// <remarks>
+    /// Do not redefine <see cref="PostInitialize()"/> <see cref="OnBeginPlay()"/> or <see cref="OnEndPlay()"/> in derived classes.
+    /// <br>Instead, use <see langword="protected virtual"/> methods:</br>
+    /// <br><see cref="PostInitialize_Static()"/></br>
+    /// <br><see cref="BeginPlay_Static()"/></br>
+    /// <br><see cref="EndPlay_Static()"/></br>
+    /// <para>
+    /// To perform the initialization and cleanup: those methods are guaranteed to only be called once in the entire lifetime of the component.
+    /// </para>
+    /// </remarks>
+    public abstract class StaticActor<T> : EActor
+        where T : EActor
+    {
+        private static T instance;
+
+        /// <summary>
+        /// Gets a value indicating whether the <see cref="PostInitialize()"/> method has already been called by Unity.
+        /// </summary>
+        public static bool IsInitialized { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the <see cref="OnBeginPlay()"/> method has already been called by Unity.
+        /// </summary>
+        public static bool IsStarted { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the <see cref="OnEndPlay()"/> method has already been called by Unity.
+        /// </summary>
+        public static bool IsDestroyed { get; private set; }
+
+        /// <summary>
+        /// Gets the global access point to the unique instance of this class.
+        /// </summary>
+        public static T Instance => instance ? instance : IsDestroyed ? null : (instance = FindExistingInstance() ?? CreateNewInstance());
+
+        /// <summary>
+        /// Looks or an existing instance of the <see cref="StaticActor{T}"/>.
+        /// </summary>
+        /// <returns>The existing <typeparamref name="T"/> instance, or <see langword="null"/> if not found.</returns>
+        public static T FindExistingInstance()
+        {
+            T[] existingInstances = FindActiveObjectsOfType<T>();
+            return existingInstances == null || existingInstances.Length == 0 ? null : existingInstances[0];
```

Comment:
You can use `existingInstance?. Length is 0` instead of having the null check before